src/test/java/com/microsoft/sqlserver/jdbc/preparedStatement/BatchExecutionWithBulkCopyParseTest.java

```diff
@@ -0,0 +1,161 @@
+    }
+
+    @Test
+    public void testAll() throws SQLException, NoSuchFieldException, SecurityException, IllegalArgumentException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+        pstmt = (SQLServerPreparedStatement) connection.prepareStatement("");
+
+        String valid = "/* rando comment *//* rando comment */ INSERT /* rando comment */ INTO /* rando comment *//*rando comment*/ \"Peter\"\"\"\"Table\" /*rando comment */"
+                + " /* rando comment */ (\"c1\"/* rando comment */, /* rando comment */[c2]/* rando comment */, /* rando comment */ /* rando comment */c3/* rando comment */, c4)"
+                + "values/* rando comment */ (/* rando comment */1/* rando comment */, /* rando comment */2/* rando comment */ , '?', ?)/* rando comment */";
+
+        Field f1 = pstmt.getClass().getSuperclass().getDeclaredField("localUserSQL");
+        f1.setAccessible(true);
+        f1.set(pstmt, valid);
+
+        Method method = pstmt.getClass().getSuperclass().getDeclaredMethod("parseUserSQLForTableNameDW", boolean.class, boolean.class);
+        method.setAccessible(true);
+
+        assertEquals((String) method.invoke(pstmt, false, false), "Peter\"\"Table");
+
+        method = pstmt.getClass().getSuperclass().getDeclaredMethod("parseUserSQLForColumnListDW");
+        method.setAccessible(true);
+
+        ArrayList<String> columnList = (ArrayList<String>) method.invoke(pstmt);
+        ArrayList<String> columnListExpected = new ArrayList<String>();
+        columnListExpected.add("c1");
+        columnListExpected.add("c2");
+        columnListExpected.add("c3");
+        columnListExpected.add("c4");
+
+        for (int i = 0; i < columnListExpected.size(); i++) {
+            assertEquals(columnList.get(i), columnListExpected.get(i));
+        }
+
+        method = pstmt.getClass().getSuperclass().getDeclaredMethod("parseUserSQLForValueListDW", boolean.class);
+        method.setAccessible(true);
+
+        ArrayList<String> valueList = (ArrayList<String>) method.invoke(pstmt, false);
+        ArrayList<String> valueListExpected = new ArrayList<String>();
+        valueListExpected.add("1");
+        valueListExpected.add("2");
+        valueListExpected.add("'?'");
+        valueListExpected.add("?");
+
+        for (int i = 0; i < valueListExpected.size(); i++) {
+            assertEquals(valueList.get(i), valueListExpected.get(i));
+        }
+    }
+
+    @BeforeEach
+    public void testSetup() throws TestAbortedException, Exception {
+        assumeTrue(13 <= new DBConnection(connectionString).getServerVersion(),
+                "Aborting test case as SQL Server version is not compatible with Always encrypted ");
+
+        connection = DriverManager.getConnection(connectionString);
+        SQLServerStatement stmt = (SQLServerStatement) connection.createStatement();
+        Utils.dropTableIfExists("esimple", stmt);
```

Comment:
Consider a UID or something, we might be dropping user table here. Also, doesn't the test use Peter_Table, why is it `esimple` here.

Reply:
Same as above. I removed the esimple part, that was unnecessary.

Reply:
We can remove this connection stuff altogether if we don't need a database for these tests.

Reply:
We need to initialize the connection object (and remove the rest, which I did), since we need the connection object to create the statement.